pkg/testutil/request_helpers.go

```diff
@@ -167,13 +167,13 @@ func (c *CompletedRequest) UnmarshalBodyToObject(obj interface{}) error {
 
 	// Content type can have an annotation after ;
 	contentParts := strings.Split(ctype, ";")
-
-	switch strings.TrimSpace(contentParts[0]) {
-	case "application/json":
-		return json.Unmarshal(c.Recorder.Body.Bytes(), obj)
-	default:
-		return fmt.Errorf("no Content-Type on response")
+	content := strings.TrimSpace(contentParts[0])
+	handler := getHandler(content)
+	if handler == nil {
+		return fmt.Errorf("unhandled content: %s", content)
 	}
+
+	return handler(c.Recorder.Body.Bytes(), obj)
```

Comment:
Could you just pass the io.ReadCloser directly, instead of bytes() to the handler? It allows it to do different things inside.